src/compute/src/server.rs

```diff
@@ -128,9 +145,10 @@ pub async fn compute_node_serve(
     let mut join_handle_vec = vec![];
 
     let state_store = StateStoreImpl::new(
-        &config.storage.state_store,
+        state_store_url,
         &config.storage.file_cache.dir,
         &config,
+        &system_params,
```

Comment:
What about making a type alias of `SystemParamsRef = Arc<..>` and directly passing the `Arc` everywhere?